### PR TITLE
feat: carbonaut configuration package init

### PR DESCRIPTION
**Pull request by leonardpahlke**
Signed-off-by: leonardpahlke <leonard.pahlke@googlemail.com>

## Proposed changes

Add a the package `pkg/config` to manage the carbonaut configuration. The configuration is 

New libraries:
* [viper](https://github.com/spf13/viper) to parse configuration
* [go-defaults](https://github.com/mcuadros/go-defaults) to set defaults in structs
* [go-validator](https://github.com/go-validator/validator) to check if struct val's are correct

Goals: 
* It should be quite easy to extend the carbonaut configuration. 
* Different kinds of configuration types are supported in the future

> It should be quite easy to extend the carbonaut configuration

Packages that need to get configured like `pkg/api` can get exteneded without touching `pkg/config` (only the inital reference must happen how the api config looks like does not matter)

> Different kinds of configuration types are supported in the future

Viper supports all sorts of different options for configuration management...

## Types of changes

What types of changes does your code introduce to Carbonaut?

- [x] New feature (non-breaking change which adds functionality)

## Checklist

_All boxes should be checked before the PR gets merged_
- [x] I have read the [CONTRIBUTING](https://github.com/carbonaut-cloud/community/blob/main/CONTRIBUTING.md) doc
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if done in a separate PR link to it in the comments section)
- [x] I don't have a merge commit in my history (see [git merge documentation](https://www.git-scm.com/docs/git-merge#_options)) (if applicable)
- [x] I signed all my commits (see [how to sign commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits))

#### Which issue(s) this PR fixes:

Fixes #https://github.com/carbonaut-cloud/carbonaut/issues/8

## Does this PR introduce a user-facing change?

```
None
```

## Comments

Test coverage for the package 95,x% only one "option" not tested `viper.Unmarshal`

```go
func getFileConfiguration(in *FileMediumConfig) (*CarbonConfig, error) {
	// ...
	carbonConfig := CarbonConfig{}
	if err := viper.Unmarshal(&carbonConfig); err != nil {
		return nil, fmt.Errorf("unable to decode into struct, %v", err)
	}
	return &carbonConfig, nil
}
```
Any idea how this can get triggered? It seems like that most errors are already catched if the file could not get read `viper.ReadInConfig()`... 

cc @0xE282B0

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	"carbonaut.cloud/carbonaut/pkg/api"
+	"carbonaut.cloud/carbonaut/pkg/connector"
+	"carbonaut.cloud/carbonaut/pkg/data"
+	"github.com/mcuadros/go-defaults"
+	"github.com/spf13/viper"
+	"gopkg.in/validator.v2"
+)
+
+// configMedium define how the decode the provided configuration
+type configMedium string
```
Comment (0xE282B0) [inline pkg/config/config.go:29]: The term configMedium sounds a bit unfamiliar. The viper doc uses terms like store or provider.

**Review by 0xE282B0 — APPROVED** [review]
LGTM

File: pkg/config/config.go (hunk 1 above)
Comment (leonardpahlke) [reply pkg/config/config.go:29]: true, I will fix that

**Review by 0xE282B0 — APPROVED** [review]

**Pull request merged**